e2e/user/user-can-cancel-edit-account.spec.ts

```diff
@@ -0,0 +1,31 @@
+import { test, expect } from "@playwright/test";
+import { loginKnownUser } from "../_lib/pom/init/login-known-user";
+import { gotoUserMyAccountPage } from "../_lib/sequences/nav-gotos";
+import { UserMyAccountEditPage } from "../_lib/pom/pages/user-my-account-edit-page";
+
+test("user can login, go to edit page, but cancel, and should not see their details changed", async ({
+  page,
+}) => {
+  // GIVEN
+  const { context, knownUser } = await loginKnownUser({ page });
+  const accountPage = await gotoUserMyAccountPage({ context });
+  const { userName, userEmail, userPhoneNumber, userResidency } = knownUser;
+  await expect(accountPage.exactText(userName)).toBeVisible();
+  await expect(accountPage.exactText(userEmail)).toBeVisible();
+  await expect(accountPage.exactText(userPhoneNumber)).toBeVisible();
+  await expect(accountPage.exactText(userResidency)).toBeVisible();
+
+  // WHEN user clicks on edit button, navigate to edit page
+  await accountPage.editButton().click();
+  const editAccountPage = new UserMyAccountEditPage(context);
+  await editAccountPage.checkUrl();
+
+  // BUT cancelled, navigate back to account page
+  await editAccountPage.cancelButton().click();
+
+  // THEN
+  await expect(accountPage.exactText(userName)).toBeVisible();
+  await expect(accountPage.exactText(userEmail)).toBeVisible();
+  await expect(accountPage.exactText(userPhoneNumber)).toBeVisible();
+  await expect(accountPage.exactText(userResidency)).toBeVisible();
```

Comment:
Due to delays in page changes, it is possible for these assertions to execute before the navigation is completed and pass based on text found on the editAccountPage. Thus we should start the THEN section with an accountPage.checkUrl to ensure these assertions execute against values on the accountPage.

Reply:
Addressed 76415d0b7a43453a12a8e9f100ff2ea5c5349abf